common/remote/sockfile.cpp

```diff
@@ -2927,11 +2927,12 @@ inline void appendErr3(MemoryBuffer &reply, RemoteFileCommandType e, int code, c
 
 
 
-#define MAPCOMMAND(c,p) case c: { ret =  this->p(msg, reply) ; break; }
-#define MAPCOMMANDCLIENT(c,p,client) case c: { ret = this->p(msg, reply, client); break; }
-#define MAPCOMMANDCLIENTTHROTTLE(c,p,client,throttler) case c: { ret = this->p(msg, reply, client, throttler); break; }
-#define MAPCOMMANDSTATS(c,p,stats) case c: { ret = this->p(msg, reply, stats); break; }
-#define MAPCOMMANDCLIENTSTATS(c,p,client,stats) case c: { ret = this->p(msg, reply, client, stats); break; }
+#define DEFINESTATS(c) ClientStats *stats = clientStatsTable.addClientReference(c, client->queryPeerName())
+#define MAPCOMMAND(c,p) case c: { DEFINESTATS(c); ret = this->p(msg, reply) ; break; }
+#define MAPCOMMANDCLIENT(c,p,client) case c: { DEFINESTATS(c); ret = this->p(msg, reply, client); break; }
+#define MAPCOMMANDCLIENTTHROTTLE(c,p,client,throttler) case c: { DEFINESTATS(c); ret = this->p(msg, reply, client, throttler); break; }
```

Comment:
Originally that's what I was going to do, but preferred not calling it at all for invalid cmds..
So that's why it's in the case statements themselves now (via the macros)